lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/failed.yaml

```diff
@@ -25,6 +25,20 @@ states:
             targetState: RETURN_TO_RP
 
   FAILED_CONFIRM_DETAILS:
+    events:
+      next:
+        targetJourney: FAILED
+        targetState: FAILED_CONFIRM_DETAILS_WITHOUT_DELETION
```

Comment:
This is pretty sneaky - 'switching journey' to a state within the same journey(!) This probably has a couple of unintended side-effects, like recording multiple SUBJOURNEY_START audit events and showing up weirdly in the journey map that we need to think through.

I think we can just do it in one state instead? It's long, but I think that's necessary - there are 4 different possible targets depending on the situation:
```
  FAILED_CONFIRM_DETAILS:
    events:
      next:
        targetState: CRI_TICF_BEFORE_COULD_NOT_CONFIRM_DETAILS
        auditEvents:
          - IPV_USER_DETAILS_UPDATE_END
        auditContext:
          successful: false
        checkIfDisabled:
          ticf:
            targetState: COULD_NOT_CONFIRM_DETAILS_PAGE
            auditEvents:
              - IPV_USER_DETAILS_UPDATE_END
            auditContext:
              successful: false
        checkFeatureFlag:
          updateDetailsAccountDeletion:
            targetState: CRI_TICF_BEFORE_COULD_NOT_CONFIRM_DETAILS_WITH_DELETION
            auditEvents:
              - IPV_USER_DETAILS_UPDATE_END
            auditContext:
              successful: false
            checkIfDisabled:
              ticf:
                targetState: COULD_NOT_CONFIRM_DETAILS_PAGE_WITH_DELETION
                auditEvents:
                  - IPV_USER_DETAILS_UPDATE_END
                auditContext:
                  successful: false
```

Of course once we remove the feature flag it'll simplify back down again :)

Reply:
ah ok - didn't realise you could combine the checkFeatureFlag and checkIfDisabled checks together, will work on this tomorrow :)

Reply:
No worries - it's not super obvious (or beautiful...) but I think conveys the intent relatively well.

One important thing to know is that disabled checks take precedence over feature flags (and otherwise it's in the order written).